cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -168,9 +170,12 @@ public void initialize(TwillContext context) {
       metricsCollectionService = injector.getInstance(MetricsCollectionService.class);
       streamCoordinatorClient = injector.getInstance(StreamCoordinatorClient.class);
 
+      programOpts = createProgramOptions(cmdLine, context, context.getSpecification().getConfigs());
+
       // Initialize log appender
       logAppenderInitializer = injector.getInstance(LogAppenderInitializer.class);
       logAppenderInitializer.initialize();
+      SystemArguments.setLogLevel(resolveScope(programOpts), logAppenderInitializer);
```

Comment:
We should use the `resolveScope` when constructing `programOpts` above (inside the `createProgramOptions` method), so that we just use `programOpts.getUserArgument()` here.